Package.swift

```diff
@@ -16,13 +16,15 @@ let package = Package(
         MANGLE_END */
     ],
     dependencies: [
-        .package(url: "https://github.com/apple/swift-crypto.git", "1.0.0" ..< "3.0.0")
+        .package(url: "https://github.com/apple/swift-crypto.git", "1.0.0" ..< "3.0.0"),
+		.package(url: "https://github.com/tsolomko/SWCompression.git", from: "4.7.0")
```

Comment:
I'd prefer a dependency that's closer to the community standards such as [Adam Fowler's](https://github.com/adam-fowler/compress-nio/blob/main/Package.swift) gzip/deflate support. Adam is a member of the SSWG, which makes me trust the maintenance of the package more. Since Vapor/JWT-Kit is a critical and widely used library, I'm not keen on adding dependencies.

Is there a way we could add compression support using a protocol - leaving the compression algorithms themselves up to other packages? That way we wouldn't have to introduce new dependencies.

NIT: Spacing

Reply:
What @Joannis said - I'm not comfortable requiring a dependency that's not either one of the Apple packages or part of the SSWG collection.

That being said, while a protocol is a good idea, it would also be desirable to provide a default implementation (based upon something that fits the aforementioned criteria for a "safe" dependency).

Reply:
Oh, I hadn’t seen that library at the time, I’ll go take a look! I’d advocate for Gwynne’s approach here, since deflate is really the only compressed JWT type in use at the moment. If adding another dependency (even a safe one) is problematic for this, that could be something like a JWTCompression library with like 10 lines.